### PR TITLE
Convert trieNode to ES6 class

### PR DESCRIPTION
**Pull request by s1na**
This PR converts `TrieNode` to an ES6 class and extracts hex prefix encoding functions to a different file, as hex prefix is an encoding method and is conceptually separate from a trie node (although encoding is used in the trie node class). It also transforms source files with `babelify` for karma tests.

If I'm not mistaken, hex prefix functions were not exposed and therefore this shouldn't be an API change.

### CODE REVIEW

**Comment by holgerd77** [comment]
Can we name this `hex-prefix.js` file differently - probably just `util.js` - since the name is not describing the range of functionality brought together there?

**Comment by holgerd77** [comment]
Update: just saw that there was a file `util.js` before, still think it would be better to combine there (so: under that name). Or do you have got an alternative suggestion for a more fitting name?

**Comment by s1na** [comment]
@holgerd77 I agree the name is a bit confusing (can be confused with `0x` hex prefix), but this is the name the yellow paper (appendix c) gives to this encoding method. The functions in `hex-prefix.js` are are related to [encoding paths](https://github.com/ethereum/wiki/wiki/Patricia-Tree#specification-compact-encoding-of-hex-sequence-with-optional-terminator), decoding them, comparing them, etc.

I'll try to make this more clear during typescript transition by using types.

**Comment by holgerd77** [comment]
One first-thought suggestion: we could add a directory `util` with three files `execution.js` (what has been left in `util.js`), `hex.js` (the two hex-functions) and `nibbles.js` (nibble-related functionality).

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "test": "npm run test:node && npm run test:browser",
-    "coverage": "istanbul cover tape ./test/*.js",
+    "coverage": "istanbul cover babel-tape-runner ./test/*.js",
```
[the remark this answers is not in the document]
Comment (s1na) [reply package.json]: The coverage tests were failing due to some ES6 syntax. Unit tests were already using `babel-tape-runner` (`"test:node": "babel-tape-runner ./test/*.js",`).

[the remark this answers is not in the document]
Comment (holgerd77) [reply package.json]: Ah ok.

File: src/hex-prefix.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    val = val.slice(1)
+  } else {
+    val = val.slice(2)
+  }
+
+  return val
+}
+
+/**
+ * Determines if a key has Arnold Schwarzenegger in it.
```
[the remark this answers is not in the document]
Comment (s1na) [reply src/hex-prefix.js]: Haha, yeah I had the same feeling when I saw this :)) Added a description.

**Pull request merged**